### PR TITLE
DAT-12406. upgraded test-harness version to 1.0.8. added LiquibaseHarnessFoundationalCompatibilitySuite

### PR DESCRIPTION
**Pull request by PavloTytarchuk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,23 @@
             <artifactId>liquibase-core</artifactId>
             <version>${liquibase.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-commercial</artifactId>
+            <!-- This syntax means open-ended version range. Take newest accessible starting from 4.14.0 -->
```
Comment (KushnirykOleh) [inline pom.xml:85]: remove this comment, it doesn't make sense here. Extension doesn't use newest available version, but specific one

File: src/test/resources/liquibase/harness/change/changelogs/yugabytedb/addAutoIncrement.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (KushnirykOleh) [inline src/test/resources/liquibase/harness/change/changelogs/yugabytedb/addAutoIncrement.xml:1]: i don't think we need to override all these changelog files, only those that are different for this platform.
Same for expectedSnapshot. If default works, let's use it.
Only expectedSQL need to used for every platform separately

**Review by KushnirykOleh — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (PavloTytarchuk) [reply pom.xml]: removed.

File: src/test/resources/liquibase/harness/change/changelogs/yugabytedb/addAutoIncrement.xml (hunk 2 above)
Comment (PavloTytarchuk) [reply src/test/resources/liquibase/harness/change/changelogs/yugabytedb/addAutoIncrement.xml]: removed all redundant changelolgs, Only platform specific are present now.

File: pom.xml [hunk 3]
```diff
@@ -88,14 +104,34 @@
...
+            <artifactId>postgresql</artifactId>
+            <version>42.5.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
+            <version>1.33</version>
+        </dependency>
+        <dependency>
+            <groupId>org.json</groupId>
```
Comment (KushnirykOleh) [inline pom.xml:126]: not sure if we need this dependency

**Review by KushnirykOleh — CHANGES_REQUESTED** [review]
overall looks good, the only thing i'm not sure about is json dependency

File: pom.xml (hunk 3 above)
Comment (PavloTytarchuk) [reply pom.xml]: removed

**Review by KushnirykOleh — APPROVED** [review]

**Pull request merged**